llvm/test/tools/llvm-symbolizer/debug-info-line-info.yaml

```diff
@@ -0,0 +1,175 @@
+# Test llvm-symbolizer always uses line info from debug info if present.
+# RUN: yaml2obj %s -o %t
+# RUN: llvm-symbolizer --obj=%t 0x1 | FileCheck %s
+
+# CHECK:      foo(bool)
+# CHECK-NEXT: ??:0:0
+
```

Comment:
The code changes touch 3 paths (symbolizeCode, symbolizedInlineCode, and symbolizeData) but I guess this only tests one of those codepaths? Please add test coverage for the other two.

& perhaps some more detail in the description about the situation where this comes up
& some details about how this yaml was generated/what's interesting in the test case?

Reply:
I experimented a bit and think the dwarf info emitter (at least from llvm ir to object file) implicitly drops `DIGlobalVariable` if its line number is 0. Here's what I tried:
- `clang++ file.cpp -O2 -g  -nostdlib -emit-llvm -S -o file.ll` and then modify the line number in `DIGlobalVariable` to 0.  
- `clang++ file.ll -fuse-ld=lld -g  -shared -nostdlib -c -o file.o` to get object file.
- Using `llvm-dwarfdump --debug-info file.o` to examine the debug info location for the global variable. It's only able to show the source location for the global variable when the line in `DIGlobalVariable` is not 0.

So, I *think* the change in `symbolizeData` is unnecessary as clang toolchain doesn't produce this case (I might be wrong, it might be possible in some cases).

Reply:
Removed change on `symbolizeData`. Add test for `symbolizeCode`